src/realtime_metrics/main.py

```diff
@@ -387,7 +421,55 @@ def availability_acceptable_stop_time_updates(stop_time_updates: list[tuple[Trip
 
     return (time_slots_with_enough_updates / number_of_time_slots) * 100
 
+def availability_acceptable_vehicle_positions(vehicle_positions: list[VehiclePosition], 
+                                              time_frame_start: datetime, 
+                                              time_frame_end: datetime) -> float:
+    """
+    Computes the availability of acceptable vehicle position messages metrics for the given vehicle positions in the given time frame.
+    The metric is defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU/edit#heading=h.14woewhhbqwk. 
+    It computes the percentage of one-minute slots with two or more vehicle positions.
 
+    Parameters:
+    vehicle_positions: list of vehicle positions
+    time_frame_start: start of the time frame to observe, datetime in utc
+    time_frame_end: end of the time frame to observe, datetime in utc
+
+    Returns:
+    A float containing the percentage of one minute slots with two or more updates.
+    """
+    logger.debug("Time frame start: %s", time_frame_start)
+    logger.debug("Time frame end: %s", time_frame_end)
+
+    if len(vehicle_positions) == 0:
+        logger.info("No vehicle positions provided!")
+        return 0.0
+
+    # all vehicle positions for a given minute
+    vehicle_positions_time: dict[int, list[VehiclePosition]] = dict()
+
+    for vehicle_position in vehicle_positions:
+        timestamp: datetime = vehicle_position.timestamp.replace(tzinfo=timezone.utc)
+
+        if timestamp < time_frame_start or timestamp > time_frame_end:
+            continue
+
+        minutes_since_time_frame_start = int((timestamp - time_frame_start).total_seconds() / 60)
+        if minutes_since_time_frame_start not in vehicle_positions_time.keys():
+            vehicle_positions_time[minutes_since_time_frame_start] = []
+        vehicle_positions_time[minutes_since_time_frame_start].append(vehicle_position)
```

Comment:
Is it necessary to store the exact `vehicle position` objects for each time slot, or would it be sufficient to only store and increase a counter (as it is done in the `availability_acceptable_stop_time_updates` method)? This would reduce the memory footprint and allow for better scaling for larger time frames.